### ISSUE TITLE
Power Poles Causing Lag

### ISSUE THREAD

**Issue by Tron842** [Bilka2/AbandonedRuins#20]
While this isn't necessarily the fault of this mod, many disconnected power poles can cause a significant impact on game UPS. If it is possible to either blacklist them from being placed by this mod, or have them placed as dummy entities, that could really help,

For a bit of an example, on my current save, after running a command to delete only the poles the Electric network update went from ~8 to ~1.4.

The command I ran: 
```
/c local surface=game.player.surface
for key, entity in pairs(surface.find_entities_filtered({force="neutral", type="electric-pole"})) do
	entity.destroy()
end
```

**Comment by Bilka2** [comment]
I cannot reproduce the issue. Even when I generate ten thousand chunks the electric network takes negligible amounts of update time (~0.15ms):
![factorio-run_grATmY6JIe](https://user-images.githubusercontent.com/27060268/140641983-4d1b1e0d-ac3a-4fe7-8577-4ab033ddbc19.png)

If this kind of minimal performance impact bothers you, I would recommend that you play without any mods.

**Comment by Bilka2** [comment]
I was wondering how things scale since some of the starting chunks don't have ruins, so I generated some more chunks. This is a 25k chunk world, electric network update time is ~0.46ms:
![factorio-run_oH9eJ5ilJM](https://user-images.githubusercontent.com/27060268/140642704-16515762-5c31-4d27-991e-0350c122dd9e.png)

For comparison: An intermediate save of Anti's September subevent running at around 35 UPS, 30k chunks:
![image](https://user-images.githubusercontent.com/27060268/140642761-9a388b9b-e481-4159-82a5-9f418c5651a3.png)

Based on this additional data, I am still not concerned about the performance impact of electric poles in the ruins mod.

**Comment by SashaIndustries** [comment]
I am also looking for a feature of this I can provide my modlist if that helps?
I am wondering if its compatibility with how mods further complicate network scripts which perhaps cause each separate powerpole to take up far more UPS then normal. Also Github can't upload json files O.o Will upload it marked as a TXT might need to be renamed to .json
[mod-list - Copy.txt](https://github.com/Bilka2/AbandonedRuins/files/7786725/mod-list.-.Copy.txt)
Thats amazing. Refused to upload it as a json. Only changed the file format as a rename and github was more then happy to let it upload xD

**Comment by Bilka2** [comment]
Hey, I just released version 1.1.5 which significantly reduces the amount of power poles in ruins. So the already minimal performance impact should be even further reduced.

**Comment by SashaIndustries** [comment]
Thank you very much. Also I feel I've offended you from your reponse of "Already minimal".
You didn't design the power network system or how factorio handles power so nothing of that is your fault.
I stated that with other mods it complicates power networks slowing them down so each network would take up 4-10x more time to calculate. I have mods such as power priority, Angels, Bobs, Pys, 5Dim, Liquid Science. In general all sorts of mods which further complicate it. So I wasn't blaming you for the UPS issues, only asking if there could be a fix to make it compatible with the other mods. Not looking to offend you or anything like that and if anything am grateful for you making this mod. I know I certainly am too dumb to make pretty much anything anymore due to only being awake 4-8 hours a day.
Also in case it was the last bit I said it was more aimed towards humor at Github being weird.
I've given up with most things so I have no desire to cause offense or harm to anyone.
